parsec/backend/asgi/logger.py

```diff
@@ -0,0 +1,51 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+from hypercorn.logging import Logger, AccessLogAtoms
+from typing import TYPE_CHECKING, Mapping, Iterable, Optional, Tuple, Union
+
+import base64
+import binascii
+
+if TYPE_CHECKING:
+    from hypercorn.config import Config
+    from hypercorn.typing import ResponseSummary, WWWScope
+
+
+class ParsecLogger(Logger):
+    def __init__(self, config: "Config") -> None:
+        super().__init__(config)
+
+    def atoms(
+        self, request: "WWWScope", response: "ResponseSummary", request_time: float
+    ) -> Mapping[str, str]:
+        """Create and return an access log atoms dictionary.
+        This can be overidden and customised if desired. It should
+        return a mapping between an access log format key and a value.
+        """
+        access_log = AccessLogAtoms(request, response, request_time)
+        id_value = self._find_author_header(request["headers"])
```

Comment:
`_find_author_header` is pretty simple and only called here, so no need for a function, you can even make it a one liner:
```python
author_header_value = next(v for k, v in headers if k.decode("latin1").lower() == "author", None)
```

Reply:
buuuuuut ! in fact this work is already done by AccessLogAtoms:

https://github.com/pgjones/hypercorn/blob/4645aa0e41815e2193104331c580592c6aae8dd5/src/hypercorn/logging.py#L134-L135

so we just have to do:
```python
header_author_value = access_log["authori"]
```

Reply:
Done.